Black winning condition

guilbill opened the pull request:
From now, we can win when we push the black puck out of the board.
Some graphical improvement, and some tests were added too.
![tipsy-black-winning](https://user-images.githubusercontent.com/560852/105373735-46f63880-5c07-11eb-90fb-9cb875a671e8.gif)

guillbill committed:
test(black): test winning conditions

JulienMattiussi commented on src/game.py:
```diff
@@ -1,46 +1,75 @@
-    OBSTACLE = '# '
-    EXIT = '  '
-    PUCK = {Board.RED: 'O ', Board.BLUE: '0 ', Board.BLACK: '* '}
+    TOP_PADDING = int(get_terminal_size()[1]*0.20)
```
At this time, seing the size and complexity of the game file, maybe spliting it would be useful
like 
a game.py for the rules mechanisms
a game_renderer.py for the display constant and methods

zyhou commented on src/game.py:
```diff
@@ -1,46 +1,75 @@
     def __display_winner(self, winner):
-        print(winner+" won the game! Congrats to the winner!")
+        print(Game.LEFT_PADING*' ' + winner +
+              " won the game! Congrats to the winner!")
```
I want `YOU WIN` in ascii art with some Confetti :D

JulienMattiussi commented on src/game.py:
```diff
@@ -1,46 +1,75 @@
-        self.pucks = {Board.RED: 6, Board.BLUE: 6}
+        self.we_have_a_winner = False
+        self.current_player = Board.RED
+        self.pucks = {Board.RED: 6, Board.BLUE: 6, Board.BLACK: 1}
```
Can you initialize this with actual length of the initial pucks arrays ?
To ensure it is always consistent

guilbill replied:
even better I think, I'm going to add a method on Board to get a count of the remaining puck of a certain color, this way it will always be consistent.

JulienMattiussi commented on src/test_game.py:
```diff
@@ -1,30 +1,77 @@
+
+        # THEN
+        self.assertEqual(game.we_have_a_winner, Board.BLUE)
+
```
Just one more test to ensure blue is not always the winner

guillbill committed:
refacto(pucks count): add puck count to Board

guillbill committed:
test(red win) red win when no red puck left

guillbill committed:
refacto(gameRenderer) split game.py

Split game.py to keep game logic into Game, and move rendering logic into new GameRenderer

guillbill committed:
feat(UX): improve UX with clearer game infos

guillbill committed:
feat(UX): add color to player name on winning message
